### DISCUSSION TITLE
dropout_adj

### DISCUSSION THREAD

**Discussion by DuanhaoranCC** [pyg-team/pytorch_geometric#4519] [General]
Hello, I suggest that ```from torch_geometric.utils import dropout_adj, to_dense_adj``` support SparseTensor(data.adj_t),it only support data.edge_index now, I want to get the augmentation adjacency matrix. Do you have any suggestions? Thank you very much for your answer! In addition, it is desirable to increase the conversion from adjacency matrix to edge_index to better create Data.
I found that when I process PPI datasets on Linux, the data is:
```python
DataBatch(x=[4814, 50], edge_index=[2, 130252], y=[4814, 121], batch=[4814], ptr=[3])
```
I found that when I process PPI datasets on windows, the data is:
```python
DataBatch(x=[3865, 50], y=[3865, 121], adj_t=[3865, 3865, nnz=103959], batch=[3865], ptr=[3])
```
This makes it difficult for me to migrate on different machines. What is the reason?(torch-geometric 2.0.4)

```python
import torch
from torch_geometric.utils import dropout_adj, to_dense_adj, dense_to_sparse
import torch_geometric.transforms as T
from torch_geometric.datasets import PPI
from torch_geometric.loader import DataLoader
from torch_geometric.nn import GCN2Conv

path = "./"
pre_transform = T.Compose([T.GCNNorm()])
train_dataset = PPI(path, split='train')
train_loader = DataLoader(train_dataset, batch_size=2, shuffle=True)
for data in train_loader:
      data = data.to(device)
      edge_index_1 = dropout_adj(data.adj_t, p=0.3)[0]
'''

**Comment by Padarn** [comment]
The difference between Windows and Linux is quite strange, could you check the source of `PPI` on the windows machine, and maybe provide a full sample?

**Reply by DuanhaoranCC** [reply]
```python
import torch

import torch_geometric.transforms as T
from torch_geometric.datasets import PPI
from torch_geometric.loader import DataLoader

path = "./"
pre_transform = T.Compose([T.GCNNorm(), T.ToSparseTensor()])
train_dataset = PPI(path, split='train', pre_transform=pre_transform)
val_dataset = PPI(path, split='val', pre_transform=pre_transform)
test_dataset = PPI(path, split='test', pre_transform=pre_transform)
train_loader = DataLoader(train_dataset, batch_size=2, shuffle=True)
val_loader = DataLoader(val_dataset, batch_size=2, shuffle=False)
test_loader = DataLoader(test_dataset, batch_size=2, shuffle=False)
for data in train_loader:
    print(data)
```
Linux:
```python
DataBatch(x=[5041, 50], edge_index=[2, 140426], y=[5041, 121], batch=[5041], ptr=[3])
```
Windows:
```python
DataBatch(x=[2844, 50], y=[2844, 121], adj_t=[2844, 2844, nnz=65390], batch=[2844], ptr=[3])
```
The result of different operation systems is different, so I can't run the same code on different machines. This is all my code, when I print ```data``` I have the above question.

**Reply by Padarn** [reply]
This seems to be the `ToSparseTensor` doing its job. Could you try setting `remove_edge_index` to false on both platforms?
```
pre_transform = T.Compose([T.GCNNorm(), T.ToSparseTensor(remove_edge_index=False)])
```

**Reply by DuanhaoranCC** [reply]
I modified the code of the data transformation part according to your suggestion, but the result is still the same, and the same after I removed the transform.
```python
import torch

import torch_geometric.transforms as T
from torch_geometric.datasets import PPI
from torch_geometric.loader import DataLoader

path = "./"
train_dataset = PPI(path, split='train')
train_loader = DataLoader(train_dataset, batch_size=2, shuffle=True)
for data in train_loader:
    print(data)
```

**Reply by Padarn** [reply]
In the above code you posted there is no longer a `pre_transform`?

**Reply by DuanhaoranCC** [reply]
Yes, all transform remove.

**Reply by DuanhaoranCC** [reply]
```python
import torch

import torch_geometric.transforms as T
from torch_geometric.datasets import PPI
from torch_geometric.loader import DataLoader

path = "./"
pre_transform = T.Compose([T.GCNNorm(), T.ToSparseTensor(remove_edge_index=False)])
train_dataset = PPI(path, split='train', pre_transform=pre_transform)
train_loader = DataLoader(train_dataset, batch_size=2, shuffle=True)
for data in train_loader:
    print(data)
```
Windows:
```python
DataBatch(x=[4145, 50], y=[4145, 121], adj_t=[4145, 4145, nnz=109775], batch=[4145], ptr=[3])
```
Linux:
```python
DataBatch(x=[5489, 50], edge_index=[2, 154020], y=[5489, 121], batch=[5489], ptr=[3])
```
Unable to use data.edge_index.

**Reply by Padarn** [reply]
Oh, this is likely because the pre-processed data is stored on your machine. You can try removing the data that was downloaded to `./` and loading it again

**Reply by Padarn** [reply]
It should be printing a warning in this case, something like 
```
UserWarning: The `pre_transform` argument differs from the one used in the pre-processed version of this dataset. If you want to make use of another pre-processing technique, make sure to sure to delete 'test/processed' first
  f"The `pre_transform` argument differs from the one used in "```

**Reply by DuanhaoranCC** [reply]
Thank you, solved the problem, thank you very much for your patience!

**Reply by Padarn** [reply]
No worries 👍 

As for your original suggestion.. I'm not sure if we want to support this explicitly or not, but maybe yo can raise an issue if you think its still something needed?
